packages/mobile/components/ActivityList.svelte

```diff
@@ -1,44 +1,92 @@
 
     function getActivityGroupTitleForTimestamp(time: Date): string {
         const dateString = getMonthYear(time)
         return dateString === getMonthYear(new Date()) ? localize('general.thisMonth') : dateString
     }
+
+    function getActivityTileDataList(activities: Activity[]): ActivityTileData[] {
+        return activities.map((activity, index) => {
+            const currentTitle = getActivityGroupTitleForTimestamp(activity.time)
+            const previousTitle =
+                activities[index - 1] && getActivityGroupTitleForTimestamp(activities[index - 1]?.time)
+            if (currentTitle !== previousTitle || index === 0) {
+                const amount = activities.filter(
+                    (activity) => getActivityGroupTitleForTimestamp(activity.time) === currentTitle
+                ).length
+
+                return { title: currentTitle, amount, activity }
+            } else {
+                return { title: undefined, amount: undefined, activity }
+            }
+        })
+    }
+
+    function searchActivities(activities: ActivityTileData[], searchValue: string): ActivityTileData[] {
+        if (!activities || !searchValue) {
+            return activities
+        }
+
+        const searchTerm = searchValue.toLowerCase()
+        const filteredActivities = activities?.filter((activity) => {
+            const title = localize(getActivityTileTitle(activity?.activity))?.toLowerCase()
+            const transactionId = activity?.activity?.transactionId?.toLowerCase()
+
+            const addressOrName =
+                activity.activity.subject.type === 'address'
+                    ? activity.activity.subject.address.toLowerCase()
+                    : activity.activity.subject.account.name.toLowerCase()
+
+            const amount = getFormattedAmountFromActivity(
+                activity.activity.type === ActivityType.Basic || activity.activity.type === ActivityType.Foundry
+                    ? activity.activity
+                    : undefined
+            )
+                .toLowerCase()
+                .replace(getGroupSeparator(), '')
+
+            return [title, transactionId, addressOrName, amount].some((item) => item.includes(searchTerm))
+        })
+        return filteredActivities
+    }
 </script>
 
-<div class="asset-list h-full flex flex-auto flex-col flex-grow flex-shrink-0">
-    {#if activityListWithTitles.length > 0}
-        <VirtualList items={activityListWithTitles} let:item>
-            <div class="mb-2">
+<activity-list-container class="asset-list h-full flex flex-auto flex-col flex-grow flex-shrink-0">
+    <activity-list-header class="sticky pb-4">
+        <SearchInput bind:value={searchValue} />
+    </activity-list-header>
+    {#if activityTileDataList.length > 0}
+        <VirtualList items={activityTileDataList} let:item>
+            <activity-tile-container class="mb-2">
```

Comment:
the vertical spacing is not working for me 🤔

Reply:
![image](https://user-images.githubusercontent.com/3624944/218511712-75f463d2-7715-4e04-8724-a55664b175a2.png)